src/components/Evolution.ts

```diff
@@ -0,0 +1,110 @@
+import { Component } from "eightbittr/lib/Component";
+
+import { FullScreenPokemon } from "../FullScreenPokemon";
+import { IPokemon } from "./Battles";
+import { 
+    IPokemonEvolution, IPokemonEvolutionByHappiness, IPokemonEvolutionByItem, 
+    IPokemonEvolutionByLevel, IPokemonEvolutionByStats, IPokemonEvolutionByTime, 
+    IPokemonEvolutionByTrade, Pokemon 
+} from "./constants/Pokemon";
+
+/**
+ * Handles logic related to Pokemon evolution.
+ */
+export class Evolution<TGameStartr extends FullScreenPokemon> extends Component<TGameStartr> {
+    /**
+     * Holds evolution requirement checks, keyed by the method of evolution.
+     */
+    public requirementHandlers: IRequirementHandlers = {
+        level: (pokemon: IPokemon, requirements: IPokemonEvolutionByLevel): boolean => {
+            return pokemon.level >= requirements.level;
+        },
+        item: (): boolean => {
+            // Need some way to pass a flag for the item being used
+            // Item usage outside of battle does not seem to be implemented (Issue #364)
+            return false;
+        },
+        trade: (requirements: IPokemonEvolutionByTrade): boolean => {
+            // Need some way to pass a flag in that the pokemon has just been traded
+            // Currently no value for held item (Issue #439)
+            // Trading is not implemented yet (Issue #440)
+            if (requirements.item) { return false; }
+            return false;
+        },
+        happiness: (): boolean => {
+            // currently no happiness value of a Pokemon (Issue #439)
+            return false;
+        },
+        time: (): boolean => {
+            // Time of day does not seem to be implemented yet (#441)
+            return false;
+        },
+        stats: (pokemon: IPokemon, requirements: IPokemonEvolutionByStats): boolean => {
+            if (requirements.statsAreEqual) {
+                return pokemon.statistics[requirements.greaterStat].normal === pokemon.statistics[requirements.lesserStat].normal;
```

Comment:
So that you don't have duplication of code, how about storing the difference as a local variable?